src/core/Header/HeaderButton.tsx

```diff
@@ -58,10 +58,19 @@ export const HeaderButton = (props: HeaderButtonProps) => {
   } = props;
 
   useTheme();
-  const isSplitButton = menuItems && onClick;
+
+  const isSplitButton = (
+    props: Partial<SplitButtonProps>,
+  ): props is SplitButtonProps => {
+    return !!props.menuItems && !!props.onClick;
+  };
+
   const buttonProps: ButtonProps & {
     styleType: 'borderless';
+    menuItems: ((close: () => void) => JSX.Element[]) | undefined;
   } = {
+    menuItems: props.menuItems,
+    onClick: onClick,
```

Comment:
Are these two lines needed?

Reply:
apparently not. Removed. Should I add similar function to isSplitButton for DropdownButton to remove menuItems={menuItems}?

Reply:
Yeah I guess DropdownButton is fine, idk

Reply:
Weird :D so go fully same approach or none...